### PR TITLE
io raster read

### PR DESCRIPTION
**Pull request by michaeldorman**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
This is great to see! Checking out the branch on Codespaces now..

**Comment by Robinlovelace** [comment]
This is really, really great work. Many thanks @michaeldorman ! `vsicurl` prefix is powerful, which vector file formats does it work with? See some suggestions here: https://github.com/geocompx/geocompy/pull/161/files

File: 08-read-write-plot.qmd [hunk 1]
```diff
@@ -238,7 +239,46 @@ gpd.read_file(u, layer='Placemarks')
...
+
+All of the previous examples read spatial information from files stored on your hard drive. 
+However, GDAL also allows reading data directly from online resources, such as HTTP/HTTPS/FTP web resources. 
+The only thing we need to do is to add a `/vsicurl/` prefix before the path to the file. 
+Let's try it by connecting to the global monthly snow probability at 500 m resolution for the period 2000-2012 (T. Hengl 2021 add reference...). 
+Snow probability for December is stored as a Cloud Optimized GeoTIFF (COG) file (see @sec-file-formats). 
+To read an online file, we just need to provide its URL together with the `/vsicurl/` prefix:
+
+```{python}
+url = "/vsicurl/https://zenodo.org/record/5774954/files/clm_snow.prob_esacci.dec_p.90_500m_s0..0cm_2000..2012_v2.0.tif"
```
Comment (Robinlovelace) [inline 08-read-write-plot.qmd:259]: Love the use of Zenodo. Worth adding a hyperlink to the underlying research? Hopefully the URL fits on a single line, if not no big deal... We can cross that bridge when we get closer to publication of the physical book!

**Review by Robinlovelace — APPROVED** [review]

File: 08-read-write-plot.qmd [hunk 2]
```diff
@@ -238,7 +239,46 @@ gpd.read_file(u, layer='Placemarks')
...
+list(values)
+```
+
+This way, we just downloaded a single value instead of the whole, large GeoTIFF file.
+
+The above example just shows one simple (but useful) case, but there is more to explore. 
+The `/vsicurl/` prefix works not only for raster but also for vector file formats. 
+It allows reading vectors directly from online storage with `geopandas` just by adding the prefix before the vector file URL.
+
+Importantly, `/vsicurl/` is not the only prefix provided by GDAL---many more exist, such as `/vsizip/` to read spatial files from ZIP archives without decompressing them beforehand or `/vsis3/` for on-the-fly reading files available in AWS S3 buckets. You can learn more about it at <https://gdal.org/user/virtual_file_systems.html>.
```
[the remark this answers is not in the document]
Comment (michaeldorman) [reply 08-read-write-plot.qmd:281]: > Are there any ready-to-go examples? And worth signposting this or providing an example in the vector section?

Maybe we can give an example of reading a Shapefile, without downloading all files, as shown here:
https://gis.stackexchange.com/questions/402308/importing-a-shapefile-from-github-using-geopandas

We can add a Shapefile for this purpose to the `data` folder, then demonstrate how to read it from the raw GitHub contents same as in the above URL.

File: 08-read-write-plot.qmd (hunk 1 above)
Comment (michaeldorman) [reply 08-read-write-plot.qmd:259]: Sure, that's a good idea!

**Pull request merged**